Sources/MagicSDK/Core/Magic.swift

```diff
@@ -9,47 +9,84 @@ import MagicSDK_Web3
-    
+
     // MARK: - Initialization
-    
+
     /// Initialize an instance of `Magic`
     ///
     /// - Parameters:
     ///   - apiKey: Your client ID. From https://dashboard.Magic.com
     ///   - ethNetwork: Network setting
     public convenience init(apiKey: String, network: EthNetwork, locale: String = Locale.current.identifier) {
-        self.init(urlBuilder: URLBuilder(apiKey: apiKey, network: network, locale: locale))
+        self.init(urlBuilder: URLBuilder(apiKey: apiKey, locale: locale, productType: .MA))
     }
-    
+
     public convenience init(apiKey: String, customNode: CustomNodeConfiguration, locale: String = Locale.current.identifier) {
-        self.init(urlBuilder: URLBuilder(apiKey: apiKey, customNode: customNode, locale: locale))
+        let urlBuilder = URLBuilder(apiKey: apiKey, customNode: customNode, locale: locale, productType: ProductType.MA)
+        self.init(urlBuilder: urlBuilder)
     }
-    
+
     public convenience init(apiKey: String, locale: String = Locale.current.identifier) {
-        self.init(urlBuilder: URLBuilder(apiKey: apiKey, network: EthNetwork.mainnet, locale: locale))
+        self.init(urlBuilder: URLBuilder(apiKey: apiKey, locale: locale, productType: .MA))
     }
-    
+
+    /// Core constructor
     private init(urlBuilder: URLBuilder) {
-        self.overlay = WebViewController(url: urlBuilder)
-        self.rpcProvider = RpcProvider(overlay: self.overlay, urlBuilder: urlBuilder)
-        self.user = UserModule(rpcProvider: self.rpcProvider)
-        self.auth = AuthModule(rpcProvider: self.rpcProvider)
-        super.init()
+        let rpcProvider = RpcProvider(urlBuilder: urlBuilder)
+        self.user = UserModule(rpcProvider: rpcProvider)
+        self.auth = AuthModule(rpcProvider: rpcProvider)
+        super.init(rpcProvider: rpcProvider)
     }
 }
 
+/// An instance of the Magic SDK
+public class MagicConnect: MagicCore {
+
+    public let connect: ConnectModule
+
+    /// Shared instance of `Magic`
+    public static var shared: MagicConnect!
+
+    public convenience init(apiKey: String) {
+        let urlBuilder = URLBuilder(apiKey: apiKey, ethNetwork: EthNetwork.mainnet, locale: "en_US", productType: ProductType.MC)
+        self.init(urlBuilder: urlBuilder)
+    }
+
+    public convenience init(apiKey: String, network: EthNetwork) {
+        let urlBuilder = URLBuilder(apiKey: apiKey, ethNetwork: network, locale: "en_US", productType: ProductType.MC)
+        self.init(urlBuilder: urlBuilder)
+    }
+
+    private init(urlBuilder: URLBuilder) {
+        let rpcProvider = RpcProvider(urlBuilder: urlBuilder)
+        self.connect = ConnectModule(rpcProvider: rpcProvider)
+        super.init(rpcProvider: rpcProvider)
+    }
+}
+
+public class MagicCore: NSObject {
+    
+    // MARK: - Property
+    public var rpcProvider: RpcProvider
+    
+    internal init(rpcProvider: RpcProvider) {
+        self.rpcProvider = rpcProvider
+    }
+}
+
+internal enum ProductType{
```

Comment:
Personal preference, so totally up to you, but I have found placing enums at the top of the file to be useful when reading code later. 🤷